src/ParquetMsg.chpl

```diff
@@ -1304,22 +1304,41 @@ module ParquetMsg {
           const high = saD.high;
           const locDom = sa.localSubdomain();
           var values = getGenericTypedArrayEntry(components["values"], st);
-          
+
+          segment_ct[i] += locDom.size;
+          if values.dtype == DType.Strings && locDom.size > 0 {
+            var e: SegStringSymEntry = toSegStringSymEntry(values);
+            var segStr = new SegString("", e);
+            ref ss = segStr;
+            var lens = ss.getLengths();
+            const lastOffset = if sa.size == 0 then 0 else sa[high];
+            const lastOffsetIdx = segStr.offsets.a.domain.high;
```

Comment:
I think I am able to unpack everything except the `.domain` part of this. Can i get a quick explanation of why you need it after calling the array part.

Reply:
The `.domain` is needed so that we can access the largest index of the offsets array. This is used in the case when the last offset value is the start of the last segment on the locale. Accessing this allows for localization of all offsets contained in any segments on the current locale.

Reply:
Thank you for the explanation. Looks good